distributed/shuffle/_disk.py

```diff
@@ -36,12 +37,19 @@ class DiskShardsBuffer(ShardsBuffer):
         to be processed exceeds this limit, then the buffer will block
         until below the threshold. See :meth:`.write` for the
         implementation of this scheme.
+    max_in_memory_buffer_size : int, optional
+        Size of in-memory buffer to use before flushing to disk. If
+        configured, incoming shards will first be moved to memory
```

Comment:
"moved" is somewhat misleading,  I just hold on to them (so this doesn't necessitate any copies).